libraries-apache-commons-io/src/test/java/com/baeldung/commons/io/csv/CSVReaderWriterUnitTest.java

```diff
@@ -38,9 +58,31 @@ public void givenCSVFile_whenRead_thenContentsAsExpected() throws IOException {
     }
 
     @Test
-    public void givenAuthorBookMap_whenWrittenToStream_thenOutputStreamAsExpected() throws IOException {
+    void givenCSVFile_whenReadWithEnumHeader_thenContentsAsExpected() throws IOException {
+        Reader in = new FileReader("src/test/resources/book.csv");
+
+        CSVFormat csvFormat = csvFormatBuilder.setHeader(BookHeaders.class)
+            .setSkipHeaderRecord(true)
+            .build();
+
+        Iterable<CSVRecord> records = csvFormat.parse(in);
+
+        for (CSVRecord record : records) {
+            String author = record.get(BookHeaders.author);
+            String title = record.get(BookHeaders.title);
+            assertEquals(AUTHOR_BOOK_MAP.get(author), title);
+        }
+    }
+
+    @Test
+    void givenAuthorBookMap_whenWrittenToStream_thenOutputStreamAsExpected() throws IOException {
         StringWriter sw = new StringWriter();
-        try (final CSVPrinter printer = new CSVPrinter(sw, CSVFormat.DEFAULT.withHeader(HEADERS))) {
+
+        CSVFormat csvFormat = csvFormatBuilder
```

Comment:
Let each test class completely initialise csvFormatBuilder, like
`CSVFormat.DEFAULT.builder().setHeader(BookHeaders.class).build`

Repurposing setUp method for abstracting common code logic is not a good idea. Also, since this has to be accessed in a static way I don't see a problem if each test class creates the CSVFormat from scratch.

Reply:
Sure, changes applied.